dali/test/python/test_RNNT_data_pipeline.py

```diff
@@ -0,0 +1,155 @@
+        return (dividend + divisor - 1) // divisor
+
+    def _splice_frames(self, input):
+        """
+        Frame splicing is implemented by transposing the input, padding it and reshaping.
+        Theoretically, to achieve the result now there should be one more transpose at the end,
+        but it can be skipped as an optimization
+        """
+        out = self.splicing_transpose(input)
+
+        # Because of the padding, we need to determine length of audio sample before it occurs
+        audio_len = self._div_ceil(self.get_shape(out), self.frame_splicing_factor)
+
+        out = self.splicing_pad(out)
+        out = self.splicing_reshape(out)
+        # Skipping transposing back
+        return out, audio_len
+
+    def remove_silence(self, input):
+        begin, len = self.get_nonsilent_region(input)
+        out = self.trim_silence(input, self.to_float(begin), self.to_float(len))
+        return out
+
+    def define_graph(self):
+        input, label = self.read()
+        # No resampling (aka "speed perturbation"), because of randomness of this operation
+        audio, sr = self.decode(input)
+
+        audio = self.remove_silence(audio)
+
+        # No dither because of the randomness of this operation
+        # audio = audio + self.normal_distribution(audio) * self.dither
+
+        # DALI's preemph works a little bit different than the one in native code.
+        # The difference occurs in first value in buffer.
+        audio = self.preemph(audio)
+
+        audio = self.spectrogram(audio)
+        audio = self.mel_fbank(audio)
+        audio = self.log_features(audio)
+        audio, audio_sh = self._splice_frames(audio)
+
+        # This normalization goes across ax=0, since
+        # the frame splicing doesn't transpose the tensor back
+        audio = self.normalize(audio)
+
+        # return audio.gpu(), label.gpu(), audio_sh.gpu()
+        return audio.gpu()
+
+
+def test_rnnt_data_pipeline():
+    """
+    Test compares pre-calculated output of native data pipeline with an output
+    from DALI data pipeline. There are few modification of native data pipeline
+    comparing to the reference:
+    1. All the random operations are turned off
+       (i.e. dither and presampling aka "speed perturbation")
+    2. Since DALI, as an optimization, doesn't perform one transposition in frame splicing,
+       the result from DALI pipeline has to be transposed to fit the reference data.
+    """
+    data_path = os.path.join(dali_extra_path, "db", "audio", "rnnt_data_pipeline")
+    reference_data = np.load(os.path.join(data_path, "output_data.npy"))[0]
+    pipe = RnntTrainPipeline(device_id=0, n_devices=1, file_root=data_path,
+                             file_list=os.path.join(data_path, "file_list.txt"), batch_size=1)
+    pipe.build()
+    dali_out = pipe.run()
+    output_data = dali_out[0].as_cpu().at(0)
+    output_data = np.transpose(output_data, (1, 0))
+    assert reference_data.shape == output_data.shape
```

Comment:
This is likely to fail. +/- 1 in time dimension would probably be adequate - and then compare the common part.

Reply:
Why? It's the same sample. The only thing that can impact the shape here is the nonsilence detection. And this we should assume, that gives the same results than librosa (which is in reference)